sklearnex/covariance/incremental_covariance.py

```diff
@@ -117,14 +184,63 @@ def fit(self, X, queue=None):
         self : object
             Returns the instance itself.
         """
-        n_samples, n_features = X.shape
+        self.n_samples_seen_ = 0
+
+        if sklearn_check_version("1.2"):
+            self._validate_params()
+            X = self._validate_data(X, dtype=[np.float64, np.float32], copy=self.copy)
+        else:
+            X = check_array(X, dtype=[np.float64, np.float32], copy=self.copy)
+
+        n_samples, self.n_features_in_ = X.shape
+
         if self.batch_size is None:
-            batch_size_ = 5 * n_features
+            self.batch_size_ = 5 * self.n_features_in_
```

Comment:
what is the purpose to put it to estimator's attribute?

Reply:
https://scikit-learn-enhancement-proposals.readthedocs.io/en/latest/slep010/proposal.html

Reply:
i mean batch_size_

Reply:
https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/decomposition/_incremental_pca.py#L236